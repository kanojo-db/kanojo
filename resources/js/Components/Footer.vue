<script setup lang="ts">
import { Link } from '@inertiajs/vue3';
</script>

<template>
    <footer class="footer p-10 bg-neutral text-white">
        <div>
            <img
                alt="Kanojo Logo"
                src="/images/logo-dark.svg"
                style="width: 150px; height: 48px"
                fit="contain"
            />
        </div>
        <div>
            <span class="footer-title">
                {{ $t('web.general.pages.general') }}
            </span>
            <Link
                :href="route('about.index')"
                class="link link-hover"
            >
                {{ $t('web.general.pages.about') }}
            </Link>
            <Link
                :href="route('privacy')"
                class="link link-hover"
            >
                {{ $t('web.general.pages.privacyPolicy') }}
            </Link>
            <Link
                href="#"
                class="link link-hover"
            >
                {{ $t('web.general.pages.contact_us') }}
            </Link>
            <a
                class="link link-hover"
                :href="route('scribe')"
            >
                {{ $t('web.general.pages.api') }}
            </a>
            <a
                class="link link-hover"
                href="https://kanojo1.statuspage.io/"
            >
                {{ $t('web.general.pages.status') }}
            </a>
        </div>
        <div>
            <span class="footer-title">
                {{ $t('web.general.pages.contribute') }}
            </span>
            <Link
                class="link link-hover"
                :href="route('bible.index')"
            >
                {{ $t('web.general.pages.bible') }}
            </Link>
            <Link
                class="link link-hover"
                :href="route('movies.create')"
            >
                {{ $t('web.general.pages.add_movie') }}
            </Link>
            <Link
                class="link link-hover"
                :href="route('movies.create')"
            >
                {{ $t('web.general.pages.fill_missing') }}
            </Link>
            <Link
                class="link link-hover"
                href="https://hosted.weblate.org/projects/kanojo/website/"
            >
                {{ $t('web.general.pages.help_translate') }}
            </Link>
        </div>
        <div>
            <span class="footer-title">
                {{ $t('web.general.pages.community') }}
            </span>
            <a
                class="link link-hover"
                href="https://kanojo.sleekplan.app"
            >
                {{ $t('web.general.pages.requestFeatures') }}
            </a>
            <a
                class="link link-hover"
                href="https://discord.gg/chg5KzTHHp"
            >
                {{ $t('web.general.pages.discord') }}
            </a>
            <a
                class="link link-hover"
                href="https://github.com/kanojo-db"
            >
                {{ $t('web.general.pages.github') }}
            </a>
            <a
                class="link link-hover"
                href="https://twitter.com/kanojodb"
            >
                {{ $t('web.general.pages.twitter') }}
            </a>
        </div>
    </footer>
</template>
