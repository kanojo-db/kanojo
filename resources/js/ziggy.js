const Ziggy = {"url":"http:\/\/192.168.0.44","port":null,"defaults":{},"routes":{"scribe":{"uri":"docs","methods":["GET","HEAD"]},"scribe.postman":{"uri":"docs.postman","methods":["GET","HEAD"]},"scribe.openapi":{"uri":"docs.openapi","methods":["GET","HEAD"]},"dusk.login":{"uri":"_dusk\/login\/{userId}\/{guard?}","methods":["GET","HEAD"]},"dusk.logout":{"uri":"_dusk\/logout\/{guard?}","methods":["GET","HEAD"]},"dusk.user":{"uri":"_dusk\/user\/{guard?}","methods":["GET","HEAD"]},"login":{"uri":"login","methods":["GET","HEAD"]},"logout":{"uri":"logout","methods":["POST"]},"password.request":{"uri":"forgot-password","methods":["GET","HEAD"]},"password.reset":{"uri":"reset-password\/{token}","methods":["GET","HEAD"]},"password.email":{"uri":"forgot-password","methods":["POST"]},"password.update":{"uri":"reset-password","methods":["POST"]},"register":{"uri":"register","methods":["GET","HEAD"]},"verification.notice":{"uri":"email\/verify","methods":["GET","HEAD"]},"verification.verify":{"uri":"email\/verify\/{id}\/{hash}","methods":["GET","HEAD"]},"verification.send":{"uri":"email\/verification-notification","methods":["POST"]},"user-profile-information.update":{"uri":"user\/profile-information","methods":["PUT"]},"user-password.update":{"uri":"user\/password","methods":["PUT"]},"password.confirmation":{"uri":"user\/confirmed-password-status","methods":["GET","HEAD"]},"password.confirm":{"uri":"user\/confirm-password","methods":["POST"]},"horizon.stats.index":{"uri":"horizon\/api\/stats","methods":["GET","HEAD"]},"horizon.workload.index":{"uri":"horizon\/api\/workload","methods":["GET","HEAD"]},"horizon.masters.index":{"uri":"horizon\/api\/masters","methods":["GET","HEAD"]},"horizon.monitoring.index":{"uri":"horizon\/api\/monitoring","methods":["GET","HEAD"]},"horizon.monitoring.store":{"uri":"horizon\/api\/monitoring","methods":["POST"]},"horizon.monitoring-tag.paginate":{"uri":"horizon\/api\/monitoring\/{tag}","methods":["GET","HEAD"]},"horizon.monitoring-tag.destroy":{"uri":"horizon\/api\/monitoring\/{tag}","methods":["DELETE"],"wheres":{"tag":".*"}},"horizon.jobs-metrics.index":{"uri":"horizon\/api\/metrics\/jobs","methods":["GET","HEAD"]},"horizon.jobs-metrics.show":{"uri":"horizon\/api\/metrics\/jobs\/{id}","methods":["GET","HEAD"]},"horizon.queues-metrics.index":{"uri":"horizon\/api\/metrics\/queues","methods":["GET","HEAD"]},"horizon.queues-metrics.show":{"uri":"horizon\/api\/metrics\/queues\/{id}","methods":["GET","HEAD"]},"horizon.jobs-batches.index":{"uri":"horizon\/api\/batches","methods":["GET","HEAD"]},"horizon.jobs-batches.show":{"uri":"horizon\/api\/batches\/{id}","methods":["GET","HEAD"]},"horizon.jobs-batches.retry":{"uri":"horizon\/api\/batches\/retry\/{id}","methods":["POST"]},"horizon.pending-jobs.index":{"uri":"horizon\/api\/jobs\/pending","methods":["GET","HEAD"]},"horizon.completed-jobs.index":{"uri":"horizon\/api\/jobs\/completed","methods":["GET","HEAD"]},"horizon.silenced-jobs.index":{"uri":"horizon\/api\/jobs\/silenced","methods":["GET","HEAD"]},"horizon.failed-jobs.index":{"uri":"horizon\/api\/jobs\/failed","methods":["GET","HEAD"]},"horizon.failed-jobs.show":{"uri":"horizon\/api\/jobs\/failed\/{id}","methods":["GET","HEAD"]},"horizon.retry-jobs.show":{"uri":"horizon\/api\/jobs\/retry\/{id}","methods":["POST"]},"horizon.jobs.show":{"uri":"horizon\/api\/jobs\/{id}","methods":["GET","HEAD"]},"horizon.index":{"uri":"horizon\/{view?}","methods":["GET","HEAD"],"wheres":{"view":"(.*)"}},"sanctum.csrf-cookie":{"uri":"sanctum\/csrf-cookie","methods":["GET","HEAD"]},"ignition.healthCheck":{"uri":"_ignition\/health-check","methods":["GET","HEAD"]},"ignition.executeSolution":{"uri":"_ignition\/execute-solution","methods":["POST"]},"ignition.updateConfig":{"uri":"_ignition\/update-config","methods":["POST"]},"debugbar.openhandler":{"uri":"_debugbar\/open","methods":["GET","HEAD"]},"debugbar.clockwork":{"uri":"_debugbar\/clockwork\/{id}","methods":["GET","HEAD"]},"debugbar.assets.css":{"uri":"_debugbar\/assets\/stylesheets","methods":["GET","HEAD"]},"debugbar.assets.js":{"uri":"_debugbar\/assets\/javascript","methods":["GET","HEAD"]},"debugbar.cache.delete":{"uri":"_debugbar\/cache\/{key}\/{tags?}","methods":["DELETE"]},"welcome":{"uri":"\/","methods":["GET","HEAD"]},"privacy":{"uri":"privacy-policy","methods":["GET","HEAD"]},"login.provider":{"uri":"login\/{provider}","methods":["GET","HEAD"]},"login.provider.callback":{"uri":"login\/{provider}\/callback","methods":["GET","HEAD"]},"about.index":{"uri":"about","methods":["GET","HEAD"]},"settings.account":{"uri":"settings\/account","methods":["GET","HEAD"]},"settings.account.update":{"uri":"settings\/account","methods":["POST"]},"settings.sessions":{"uri":"settings\/sessions","methods":["GET","HEAD"]},"settings.tokens":{"uri":"settings\/tokens","methods":["GET","HEAD"]},"settings.tokens.create":{"uri":"settings\/tokens","methods":["POST"]},"settings.tokens.destroy":{"uri":"settings\/tokens\/{token}","methods":["DELETE"]},"search":{"uri":"search","methods":["GET","HEAD"]},"movies.index":{"uri":"movies","methods":["GET","HEAD"]},"movies.create":{"uri":"movies\/create","methods":["GET","HEAD"]},"movies.store":{"uri":"movies","methods":["POST"]},"movies.show":{"uri":"movies\/{movie}","methods":["GET","HEAD"]},"movies.edit":{"uri":"movies\/{movie}\/edit","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.update":{"uri":"movies\/{movie}","methods":["PUT","PATCH"],"bindings":{"movie":"slug"}},"movies.destroy":{"uri":"movies\/{movie}","methods":["DELETE"],"bindings":{"movie":"slug"}},"movies.media.index":{"uri":"movies\/{movie}\/media","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.media.store":{"uri":"movies\/{movie}\/media","methods":["POST"],"bindings":{"movie":"slug"}},"media.destroy":{"uri":"media\/{medium}","methods":["DELETE"]},"movies.history.index":{"uri":"movies\/{movie}\/history","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.like.store":{"uri":"movies\/{movie}\/like","methods":["POST"],"bindings":{"movie":"slug"}},"movies.dislike.store":{"uri":"movies\/{movie}\/dislike","methods":["POST"],"bindings":{"movie":"slug"}},"movies.favorite.store":{"uri":"movies\/{movie}\/favorite","methods":["POST"],"bindings":{"movie":"slug"}},"favorite.destroy":{"uri":"favorite\/{favorite}","methods":["DELETE"]},"movies.wishlist.store":{"uri":"movies\/{movie}\/wishlist","methods":["POST"],"bindings":{"movie":"slug"}},"wishlist.destroy":{"uri":"wishlist\/{wishlist}","methods":["DELETE"]},"movies.collection.store":{"uri":"movies\/{movie}\/collection","methods":["POST"],"bindings":{"movie":"slug"}},"collection.destroy":{"uri":"collection\/{collection}","methods":["DELETE"]},"movies.reports.index":{"uri":"movies\/{movie}\/reports","methods":["GET","HEAD"],"bindings":{"movie":"slug"}},"movies.reports.create":{"uri":"movies\/{movie}\/reports\/create","methods":["GET","HEAD"]},"movies.reports.store":{"uri":"movies\/{movie}\/reports","methods":["POST"],"bindings":{"movie":"slug"}},"reports.show":{"uri":"reports\/{report}","methods":["GET","HEAD"]},"reports.edit":{"uri":"reports\/{report}\/edit","methods":["GET","HEAD"]},"reports.update":{"uri":"reports\/{report}","methods":["PUT","PATCH"]},"reports.destroy":{"uri":"reports\/{report}","methods":["DELETE"]},"models.index":{"uri":"models","methods":["GET","HEAD"]},"models.create":{"uri":"models\/create","methods":["GET","HEAD"]},"models.store":{"uri":"models","methods":["POST"]},"models.show":{"uri":"models\/{model}","methods":["GET","HEAD"],"bindings":{"model":"id"}},"models.edit":{"uri":"models\/{model}\/edit","methods":["GET","HEAD"],"bindings":{"model":"id"}},"models.update":{"uri":"models\/{model}","methods":["PUT","PATCH"],"bindings":{"model":"id"}},"models.destroy":{"uri":"models\/{model}","methods":["DELETE"]},"models.media.index":{"uri":"models\/{model}\/media","methods":["GET","HEAD"],"bindings":{"model":"id"}},"models.media.store":{"uri":"models\/{model}\/media","methods":["POST"],"bindings":{"model":"id"}},"models.history.index":{"uri":"models\/{model}\/history","methods":["GET","HEAD"],"bindings":{"model":"id"}},"studios.index":{"uri":"studios","methods":["GET","HEAD"]},"studios.create":{"uri":"studios\/create","methods":["GET","HEAD"]},"studios.store":{"uri":"studios","methods":["POST"]},"studios.show":{"uri":"studios\/{studio}","methods":["GET","HEAD"],"bindings":{"studio":"id"}},"studios.edit":{"uri":"studios\/{studio}\/edit","methods":["GET","HEAD"],"bindings":{"studio":"id"}},"studios.update":{"uri":"studios\/{studio}","methods":["PUT","PATCH"],"bindings":{"studio":"id"}},"studios.destroy":{"uri":"studios\/{studio}","methods":["DELETE"],"bindings":{"studio":"id"}},"profile.show":{"uri":"user\/{user}","methods":["GET","HEAD"],"bindings":{"user":"id"}},"bible.general":{"uri":"bible\/general","methods":["GET","HEAD"]},"user.locale.store":{"uri":"user\/locale","methods":["POST"]}}};

if (typeof window !== 'undefined' && typeof window.Ziggy !== 'undefined') {
    Object.assign(Ziggy.routes, window.Ziggy.routes);
}

export { Ziggy };
