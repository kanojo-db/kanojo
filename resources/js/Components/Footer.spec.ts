import { createTestingPinia } from '@pinia/testing';
import { renderToString, shallowMount } from '@vue/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import Footer from '@/Components/Footer.vue';
import getI18nPlugin from '@/plugins/i18n';
import getVuetifyPlugin from '@/plugins/vuetify';
import { ziggyProps } from '@/test/ziggy.mock';

expect.extend(toHaveNoViolations);

const i18n = getI18nPlugin('en-US');
const vuetify = getVuetifyPlugin(i18n);

vi.mock('@inertiajs/vue3', async () => {
    const mod =
        await vi.importActual<typeof import('@inertiajs/vue3')>(
            '@inertiajs/vue3',
        );

    return {
        ...mod,
        usePage: () => ({
            props: {
                value: {
                    ...ziggyProps,
                },
            },
        }),
    };
});

describe('BtnDropdown', () => {
    it('renders properly in SSR mode', async () => {
        const contents = await renderToString(
            {
                template: '<v-app><v-layout><Footer /></v-layout></v-app>',
                components: {
                    Footer,
                },
            },
            {
                global: {
                    plugins: [i18n, vuetify, createTestingPinia()],
                    mocks: {
                        page: {
                            props: {
                                ...ziggyProps,
                            },
                        },
                        $page: {
                            component: 'Welcome',
                        },
                    },
                },
            },
        );

        expect(contents).toMatchSnapshot();
    });

    it('has no a11y violations', async () => {
        const wrapper = shallowMount(Footer);

        expect(await axe(wrapper.html())).toHaveNoViolations();
    });
});
