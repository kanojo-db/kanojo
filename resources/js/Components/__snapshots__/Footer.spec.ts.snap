// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BtnDropdown > renders properly in SSR mode 1`] = `"<div class=\\"v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr\\" style=\\"z-index:1000;\\"><div class=\\"v-application__wrap\\"><!--[--><div class=\\"v-layout\\" style=\\"z-index:900;position:relative;overflow:hidden;\\"><!--[--><footer class=\\"v-footer v-theme--light grow-0 flex-col items-start justify-center gap-4 bg-stone-600 p-10 text-stone-50 dark:bg-stone-800 md:flex-row md:gap-16\\" style=\\"height:auto;\\"><div><img alt=\\"Kanojo Logo\\" src=\\"/images/logo-dark.svg\\" class=\\"h-[48px]] w-[150px] object-contain\\"></div><div class=\\"flex flex-col gap-4 md:flex-row md:gap-36\\"><div class=\\"flex flex-col\\"><span class=\\"mb-2 font-bold\\">General</span><a class=\\"hover:text-secondary\\" href=\\"http://192.168.0.44/about\\">About Kanojo</a><a class=\\"hover:text-secondary\\" href=\\"http://192.168.0.44/plugins\\">Plugins</a><a class=\\"hover:text-secondary\\" href=\\"http://192.168.0.44/privacy-policy\\">Privacy Policy</a><a href=\\"http://developer.kanojodb.com\\" class=\\"hover:text-secondary\\">API</a></div><div class=\\"flex flex-col\\"><span class=\\"mb-2 font-bold\\">Contribute</span><a class=\\"hover:text-secondary\\" href=\\"https://hosted.weblate.org/projects/kanojo/website/\\">Help Translate Kanojo</a></div><div class=\\"flex flex-col\\"><span class=\\"mb-2 font-bold\\">Community</span><a href=\\"https://kanojo.sleekplan.app\\" class=\\"hover:text-secondary\\">Request Features</a><a href=\\"https://discord.gg/chg5KzTHHp\\" class=\\"hover:text-secondary\\">Discord</a><a href=\\"https://github.com/kanojo-db\\" class=\\"hover:text-secondary\\">GitHub</a><a rel=\\"me\\" href=\\"https://mastodon.social/@kanojodb\\" class=\\"hover:text-secondary\\">Mastodon</a><a href=\\"https://twitter.com/kanojodb\\" class=\\"hover:text-secondary\\">Twitter</a></div></div></footer><!--]--></div><!--]--></div></div>"`;
